File: projects/punk0/code/rotate.js
// rotate the input list
// to the right, e.g.:
//   [1,2,3,4,5]
// returns:
//   [5,1,2,3,4]
function rotate(x) {
  let r = [];

  for (let i = 0; i < x.length; i++) {
    // go to the last element and
    // then wrap around
    let len = x.length;
    let idx = (i + len - 1) % len;
    let v = x[idx];
    r.push(v);
  }

  return r;
}

console.log(JSON.stringify(rotate([1, 1, 2, 3, 3, 4, 1, 2, 7, 9])));

File: projects/punk0/code/rotate.py
# rotate the input list to the right
#   [1,2,3,4]
# returns:
#   [4,3,2,1]
def rotate(x):
    r = []
    i = 0
    for v in x:
        # go to the last element and
        # then wrap around
        # example if len is 4:
        # (0 + 4 - 1) % 4 = 3
        # (1 + 4 - 1) % 4 = 0
        # (2 + 4 - 1) % 4 = 1
        # (3 + 4 - 1) % 4 = 2
        idx = (i + len(x) - 1) % len(x)
        v = x[idx]
        r.append(v)
        i += 1

    return r


print(rotate([1, 1, 2, 3, 3, 4, 1, 2, 7, 9]))
